front/src/modules/pathfinding/components/IncompatibleConstraints/IncompatibleConstrainstFilters.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { type FC, type HTMLAttributes } from 'react';
+
+import { Filter } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { toPairs } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import type { IncompatibleConstraintTypes } from './type';
+import { getIcon } from './utils';
+
+interface IncompatibleConstraintsFiltersProps extends HTMLAttributes<unknown> {
+  data: Record<IncompatibleConstraintTypes, { count: number; enabled: boolean }>;
+  toggle: (type: string) => void;
+}
+const IncompatibleConstraintsFilters: FC<IncompatibleConstraintsFiltersProps> = (props) => {
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+  const { data, toggle, ...htmlAttrs } = props;
+
+  return (
+    <div {...htmlAttrs}>
+      <Filter title={t('incompatibleConstraints.filters.title')} />
+      {toPairs(data)
+        .filter(([, type]) => type.count > 0)
+        .map(([name, type]) => {
+          const Icon = getIcon(name as IncompatibleConstraintTypes);
+          const label = t(`incompatibleConstraints.${name}`).toString();
+          return (
+            <button
+              key={name}
+              type="button"
+              title={
+                type.enabled
+                  ? t('incompatibleConstraints.filters.disable', { name: label })
```

Comment:
Better after your fix, but there is still one issue : 

![Capture d’écran 2024-08-05 à 17 53 49](https://github.com/user-attachments/assets/d46c61dd-dbfe-43d8-8737-2a5d947e9ddf)

Here if I click on the signal icon, it will activate all filters again but it says "hide signaling filters"